### PR TITLE
Updated Dev Container

### PR DESCRIPTION
**Pull request by malcmiller**
Updated the dev-container with More extensions to help local dev, added more pip requirements necessary for the custom model development, and added pip caching.

### CODE REVIEW

**Review by marshallbentley — APPROVED** [review]

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
+FROM python:3.12-bullseye
+
+RUN set -x
+
+RUN apt-get update \
+    && apt-get -y install curl \
+    && apt-get -y install net-tools \
+    && apt-get -y install procps \
+    && apt-get -y install build-essential \
+    && apt-get -y install docker.io
```
Comment (kottofy) [inline .devcontainer/Dockerfile:10]: Just wondering if we could pull these all in one command.. something like this?
```
RUN apt-get update \
    && apt-get -y install \
        curl \
        net-tools \
        procps \
        build-essential \
        docker.io
```

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
+				"formulahendry.code-runner",
+				"redhat.vscode-yaml",
+				"streetsidesoftware.code-spell-checker",
+				"ms-python.black-formatter",
+				"ms-python.isort"
+			]
+		}
+	},
+	"runArgs": ["-v", "/var/run/docker.sock:/var/run/docker.sock"],
+	"postStartCommand": "pre-commit install"
```
Comment (kottofy) [inline .devcontainer/devcontainer.json:33]: I think it's recommended to not have pre-commit installed by default and instead instruct the developers on how to install it manually.

[Playbook reference](https://microsoft.github.io/code-with-engineering-playbook/code-reviews/recipes/python/#pre-commit-hooks)

image

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (malcmiller) [reply .devcontainer/Dockerfile:10]: Yeah I didn't change that but can done that way!

Comment (malcmiller) [reply .devcontainer/Dockerfile]: updated

**Pull request merged**